asabiris/orchestration/sendemail.py

```diff
@@ -6,206 +6,164 @@
-            )
-
-        return error_message, "Error Processing Request"
+	"""
+	A class to orchestrate the sending of emails.
+
+	This class handles rendering email templates, processing attachments, and
+	sending emails through an SMTP service.
+
+	"""
+
+	def __init__(self, app):
+		"""
+		Initialize the SendEmailOrchestrator with necessary services.
+
+		Args:
+			app: The application object, used to get services.
+		"""
+		self.JinjaService = app.get_service("JinjaService")
+		self.MarkdownToHTMLService = app.get_service("MarkdownToHTMLService")
+		self.AttachmentRenderingService = app.get_service("AttachmentRenderingService")
+
+		self.SmtpService = app.get_service("SmtpService")
+
+
+	async def send_email(
+		self,
+		email_to: List[str],
+		body_template: str,
+		body_params=None,
+		email_from=None,
+		email_cc=None,
+		email_bcc=None,
+		email_subject=None,
+		attachments=None
+	):
+		"""
+		Send an email using specified parameters.
+		...
+		"""
+		try:
+			body_params = body_params or {}
+			attachments = attachments or []
+			email_cc = email_cc or []
+			email_bcc = email_bcc or []
+
+			# Rendering the template
+			body_html, email_subject_body = await self._render_template(body_template, body_params)
+
+			# If email_subject is not provided or is empty use email_subject_body
+			if email_subject is None or email_subject == '':
+				email_subject = email_subject_body
+
+			# Processing attachments
+			atts_gen = self.AttachmentRenderingService.render_attachment(attachments)
+
+			# Sending the email
+			await self.SmtpService.send(
+				email_from=email_from,
+				email_to=email_to,
+				email_cc=email_cc,
+				email_bcc=email_bcc,
+				email_subject=email_subject,
+				body=body_html,
+				attachments=atts_gen,
+			)
+			L.info("Email sent successfully to: {}".format(', '.join(email_to)))
+
+
+		except Exception as e:
+			L.exception("Error occurred preparint the email")
+
+			error_message, error_subject = self._generate_error_message(str(e))
+			await self.SmtpService.send(
+				email_from=email_from,
+				email_to=email_to,
+				email_cc=email_cc,
+				email_bcc=email_bcc,
+				email_subject=error_subject,
+				body=error_message
+			)
+
+
+	async def _render_template(self, template: str, params: Dict) -> Tuple[str, str]:
+		if not template.startswith('/Templates/Email/'):
+			raise PathError(use_case='Email', invalid_path=template)
+
+		jinja_output = await self.JinjaService.format(template, params)
+
+		ext = os.path.splitext(template)[1]
+		if ext == '.html':
+			return find_subject_in_html(jinja_output)
+
+		elif ext == '.md':
+			body, subject = find_subject_in_md(jinja_output)
+			body = self.MarkdownToHTMLService.format(body)
+			return body, subject
+
+		else:
+			raise FormatError(format=ext)
+
+
+	def _generate_error_message(self, specific_error: str) -> Tuple[str, str]:
+		try:
+			timestamp = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+		except Exception as date_err:
```

Comment:
what could possibly go wrong in `datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")` ?